packages/big-design/src/components/Worksheet/Cell/Cell.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { memo, useCallback, useEffect, useMemo } from 'react';
+
+import { Small } from '../../Typography';
+import { useEditableCell, useStore } from '../hooks';
+import { Cell as TCell, WorksheetColumn } from '../types';
+
+import { StyledCell } from './styled';
+
+interface CellProps<Value> extends TCell<Value> {
+  validation: WorksheetColumn<Value>['validation'];
+}
+
+// We can't know what value type a cell has
+export const Cell: React.FC<CellProps<any>> = memo(({ columnIndex, hash, rowIndex, type, value, validation }) => {
```

Comment:
You can still use generics here:
```suggestion
const InternalCell = <T extends unknown>({
  columnIndex,
  hash,
  rowIndex,
  type,
  value,
  validation,
}: CellProps<T>): React.ReactElement<CellProps<T>> => {
```

And the export will be:
```tsx
export const Cell = typedMemo(InternalCell);
```

I would still try and type these with generics if you can.